pyvelox/pyvelox.h

```diff
@@ -416,6 +417,24 @@ inline void addVectorBindings(
   m.def("from_list", [](const py::list& list) mutable {
     return pyListToVector(list, PyVeloxContext::getInstance().pool());
   });
+
+  m.def("export_to_arrow", [](VectorPtr& inputVector) {
+    auto arrowArray = new ArrowArray();
+    std::shared_ptr<facebook::velox::memory::MemoryPool> pool_{facebook::velox::memory::getDefaultMemoryPool()};
+    facebook::velox::exportToArrow(inputVector, *arrowArray, pool_.get());
+    inputVector.reset();
+    return reinterpret_cast<uintptr_t>(arrowArray);
+  });
+
+  m.def(
+      "import_from_arrow",
+      [](uintptr_t arrowArrayPtr, uintptr_t arrowSchemaPtr) {
+        auto arrowArray = reinterpret_cast<ArrowArray*>(arrowArrayPtr);
+        auto arrowSchema = reinterpret_cast<ArrowSchema*>(arrowSchemaPtr);
```

Comment:
Currently, this function is written such that you pass the pointers as arguments, but alternatively, the function could accept a pyarrow.Array object, and then the pointers could be initialized here and the `_export_to_c` method could be called here? That way, the user doesn't need to call cffi themselves, and what now takes

```
        c_schema = ffi.new("struct ArrowSchema*")
        schema_ptr = int(ffi.cast("uintptr_t", c_schema))

        c_array = ffi.new("struct ArrowArray*")
        array_ptr = int(ffi.cast("uintptr_t", c_array))

        arr = pa.array([1, 2, 3], type=pa.int32())
        arr._export_to_c(array_ptr, schema_ptr)

        velox_vector = pv.import_from_arrow(array_ptr, schema_ptr)
```

could be reduced to

```
        velox_vector = pv.import_from_arrow(arr)
```

?

Reply:
Yes, that was my initial thought. But, I guess that would make PyArrow a dependency here, which I am not sure whether it will be a good approach.

Reply:
We can make it a proper dependency if thats the way to go. What would be concerns with making pyarrow a proper dependency for pyvelox ?

Reply:
> What would be concerns with making pyarrow a proper dependency for pyvelox ?

Just being an additional dependency that you need to have installed (and each additional dependency can give some installation problem), while it is not strictly needed for other functionality of PyVelox.  
Now, since we won't require a development version of pyarrow, that's shouldn't give much problems.

Reply:
Currently, the conversion function accepts a `py::obj` and calls the method `_export_to_c` internally. Does this approach look good?

cc: @jorisvandenbossche @kgpai

Reply:
@sanjibansg thanks for the update. Yes, that looks good to me! In addition, we could also do the same for the `export_to_arrow` function, so that the two methods are more each other counterparts?